### PR TITLE
Add interactive messages when extension install is in progress.

### LINKED ISSUE #7645: Add extension command seems stuck when it is taking time to install the extension
**Is your feature request related to a problem? Please describe.**
az extension add --source <>  command seems stuck when it is taking time to install a extension. There is no progress update or spinner or bar to tell me it is doing something.

**Describe the solution you'd like**
Add progress update statuses like.. getting extension, verifying extension, installing extension or atleast a spinner would be great to make the user experience better.

### PR DESCRIPTION
**Pull request by atbagga**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

Fixes #7645

### CODE REVIEW

**Comment by atbagga** [comment]
@yugangw-msft  @tjprescott I see some extensions take a lot of time to install and in interactive mode user feels the command is kind of stuck reporting no progress. This is an attempt to fix the behaviour by showing spinner and message like Downloading, Validating, Installing to let the user know whats happning in the background. 

Please let me know your thoughts on the PR?

Kapture 2019-04-02 at 2 07 32

File: src/azure-cli-core/azure/cli/core/extension/operations.py [hunk 1]
```diff
@@ -95,6 +95,7 @@ def _validate_whl_extension(ext_file):
 
 
 def _add_whl_ext(source, ext_sha256=None, pip_extra_index_urls=None, pip_proxy=None):  # pylint: disable=too-many-statements
+    from humanfriendly import AutomaticSpinner
```
Comment (yugangw-msft) [inline src/azure-cli-core/azure/cli/core/extension/operations.py]: You should getting the get a hand hold of `cli_ctx` and use `cli_ctx.get_progress_controller`

Comment (atbagga) [reply src/azure-cli-core/azure/cli/core/extension/operations.py]: @yugangw-msft  Resolved the comment. Please take look. Thanks!

**Comment by atbagga** [comment]
Kapture 2019-04-02 at 2 07 32

**Comment by atbagga** [comment]
> Before merging, we need to make sure that this change will not break anything in the extensions repo.

Sure @tjprescott, I am working on the tests right now. Let me know if I need to do anything to verify anything else.

**Comment by tjprescott** [comment]
@atbagga I did a quick check of the extensions repo and azdev, and nothing seems to rely on the signatures you changed, so I am okay with this change.

**Pull request merged**